cmake/modules/extensions.cmake

```diff
@@ -1,5 +1,13 @@
 # SPDX-License-Identifier: Apache-2.0
 
+include_guard(GLOBAL)
+
+# Extensions depends on two CMake modules, so load them here
```

Comment:
This comment demonstrates why we shouldn't be redundantly documenting what is clear from the code.

It easily gets outdated.

It should be removed.

Reply:
No, that comment is not outdated, cause the two modules below the comment are referring to official CMake modules, the `user_cache` is a Zephyr CMake module.

Reply:
Now, before I followed your other suggestion of removing the [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41301#discussion_r793504732) regarding Zephyr CMake dependencies, this looked like:
https://github.com/zephyrproject-rtos/zephyr/blob/bae4f1ba7a73103ab88563b38fc36345ff083b7f/cmake/modules/extensions.cmake#L5-L10

Which refers to two upstream CMake module dependencies and one internal Zephyr dependency.

Because I addressed you earlier comment, it is now unclear to reader what might be local Zephyr CMake modules and what are regular CMake modules, and thus where to look for them.

Reply:
perhaps it should say official/regular/upstream CMake modules instead of CMake modules then. There are three CMake modules there unless I have misunderstood the terminology.

Either that or the comment should be dropped.

Reply:
I think it's good to indicate those are CMake distributed CMake modules.

So how about:
> \# Dependencies on CMake modules from the CMake distribution.

That is following the term CMake docs is using:
https://cmake.org/cmake/help/v3.22/manual/cmake-modules.7.html

> The modules listed here are part of the CMake distribution.

afterall, C has the `"file.h"` and `<file.h>` distinction, which CMake doesn't :wink:

Reply:
sounds good

Reply:
done